deepchecks/vision/checks/performance/segment_performance.py

```diff
@@ -0,0 +1,240 @@
+        """Calculate segment performance based on image properties.
+
+        Returns
+        -------
+        CheckResult
+            value: dictionary containing performance for each property segments
+            display: table of results
+        """
+        dataset = context.get_data_by_kind(dataset_kind)
+        # In case there are fewer samples than 'number_of_samples_to_infer_bins' then bins were not calculated
+        if self._state['bins'] is None:
+            # Create the bins and metrics
+            bins = self._create_bins_and_metrics(self._state['samples_for_binning'], dataset)
+        else:
+            bins = self._state['bins']
+
+        # bins are in format:
+        # {property_name: [{start: val, stop: val, count: x, metrics: {name: metric...}}, ...], ...}
+        display_data = []
+
+        for property_name, prop_bins in bins.items():
+            for single_bin in prop_bins:
+                # If we have a low number of unique values for a property, the first bin (-inf, x) might be empty so
+                # check the count, and if empty filter out the bin
+                if single_bin['count'] == 0:
+                    continue
+
+                bin_data = {
+                    'Range': _range_string(single_bin['start'], single_bin['stop']),
+                    'Number of samples': single_bin['count'],
+                    'Property': f'Property: {property_name}'
+                }
+                # Update the metrics in the single bin from the metrics objects to metric mean results, in order to
+                # return ths bins object as the check result value
+                single_bin['metrics'] = _calculate_metrics(single_bin['metrics'], dataset)
+                # For the plotly display need row per metric in the dataframe
+                for metric, val in single_bin['metrics'].items():
+                    display_data.append({'Metric': metric, 'Value': val, **bin_data})
+
+        display_df = pd.DataFrame(display_data)
+
+        fig = px.bar(
+            display_df,
+            x='Range',
+            y='Value',
+            color='Metric',
+            barmode='group',
+            facet_col='Property',
+            facet_row='Metric',
+            facet_col_spacing=0.05,
+            facet_row_spacing=0.05,
+            hover_data=['Number of samples']
+        )
+
+        (fig.update_xaxes(title=None, type='category', matches=None)
+            .update_yaxes(title=None)
+            .for_each_annotation(lambda a: a.update(text=a.text.split('=')[-1]))
+            .for_each_yaxis(lambda yaxis: yaxis.update(showticklabels=True)))
+
+        return CheckResult(value=bins, display=fig)
+
+    def _create_bins_and_metrics(self, batch_data: t.List[t.Tuple], dataset):
+        """Return dict of bins for each property in format
+        {property_name: [{start: val, stop: val, count: x, metrics: {name: metric...}}, ...], ...}"""
```

Comment:
```suggestion
        {property_name: [{start: val, stop: val, count: x, metrics: {name: metric...}}], ...}"""
```

Reply:
this is not true, since this is a list with multiple items

Reply:
what other item is there after the metric?